### PR TITLE
Basic CMake support

### PR DESCRIPTION
**Pull request by rhodes-b**
Adds cmake as the build system

### CODE REVIEW

**Comment by rhodes-b** [comment]
Didn't update the GHA because I don't think this is done, need to figure out a better way to add the target link directories (because the install command makes `bin` `lib` dirs where you set prefix this can't be the same location as where the source is built I say the best idea is to probably move where it builds the source `libluajit.so` into CMAKE_BINARY_DIR so we get that include path for free not sure about the real lib that the target links against though

**Comment by tale** [comment]
I agree with your idea for now. I think we should add a flag like `-DLUAJIT_EMBEDDED` to statically embed LuaJIT compiled from source, otherwise fallback to looking for `luajit-2.1-whatever` in the system.

File: CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+)
+
+target_link_directories(${PROJECT} PUBLIC
+    "${CMAKE_BINARY_DIR}"
+    "${CMAKE_BINARY_DIR}/lib"
+)
+
+target_link_libraries(${PROJECT} PUBLIC
+    luajit
+    m # required for LUAJIT_EMBEDDED
```
Comment (rhodes-b) [inline CMakeLists.txt:42]: I guess this could also be wrapped in a `if(DEFINED)` block

File: CMakeLists.txt [hunk 2]
```diff
@@ -0,0 +1,46 @@
+
+cmake_minimum_required(VERSION 3.20)
```
Comment (rhodes-b) [inline CMakeLists.txt:2]: not sure what the minimum we can support is I just picked 3.20 randomly

**Comment by rhodes-b** [comment]
I think this is about the best we can do for this external

**Comment by rhodes-b** [comment]
this doesn't really support looking for luajit on your system yet I think that might require cmake to [install](https://cmake.org/cmake/help/latest/command/install.html) the static lib appropriately (install whatever is in `${CMAKE_BINARY_DIR}/lib`)

**Comment by rhodes-b** [comment]
I guess next action items are if you approve the current state would be
- [ ] remove meson stuff (mise as well)?
- [ ] update GHA

File: CMakeLists.txt [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+target_include_directories(${PROJECT} PUBLIC
+    "include/"
+    "${LUAJIT_INSTALL_PATH}/src/"
+)
+
+target_link_directories(${PROJECT} PUBLIC
+    "${CMAKE_BINARY_DIR}"
+    "${CMAKE_BINARY_DIR}/lib"
+)
```
Comment (rhodes-b) [inline CMakeLists.txt:38]: I swear I thought that cmake gave you at least CMAKE_BINARY_DIR for free but this is better than nothing

File: CMakeLists.txt (hunk 2 above)
Comment (tale) [reply CMakeLists.txt:2]: I have some tweaks I can make in this regard, also general CMake stuff that I do, but generally this `cmake_minimum_required(VERSION 3.15...3.31)`.

**Comment by tale** [comment]
> I guess next action items are if you approve the current state would be
> 
> * [ ]  remove meson stuff (mise as well)?
> * [ ]  update GHA

Not going to remove Mise. It's for dev tooling, but I'll switch it to use cmake.

File: CMakeLists.txt [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+target_link_directories(${PROJECT} PUBLIC
+     "${CMAKE_BINARY_DIR}"
+     "${CMAKE_BINARY_DIR}/lib"
+)
+
+target_link_libraries(${PROJECT} PUBLIC
+    "${LUAJIT_LIBRARIES}"
+)
+
+add_dependencies(${PROJECT} luajit-external)
```
Comment (rhodes-b) [inline CMakeLists.txt:45]: just ensures that it actually downloads the external first.

Probably not the best long term solution. Should abstract to some function or macro in a general include that takes in a target (${PROJECT}) in this case which would automatically add the dependency and link its libraries to the target so we don't have this "random" stuff like target_link_directories

Comment (rhodes-b) [reply CMakeLists.txt:45]: kinda like the find2 we found in neovims cmake

**Comment by tale** [comment]
This otherwise looks good to me, I'm going to add in some traditional "Cmake-y" things I tend to do out of habit and a GitHub Actions workflow before I merge this in but I appreciate all the work!

**Pull request merged**